tinygrad/codegen/linearizer.py

```diff
@@ -103,10 +105,11 @@ def global_load(self, i:int, idxs:List[Node], acc:Optional[LazyOp]=None, barrier
     for idx, valid, rep_idx in zip(e_idxs, e_valids, iter_idxs(expand_vars)):
       this_const, idx, valid = (invalid_value, NumNode(0), NumNode(1)) if valid.max == 0 else (const, idx, valid)
       # todo: when multiple reduceops are supported, clearly disambiguate and test acc load keys are unique for each reduceop
-      key = f"{acc is not None}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
+      key = f"{self.reduceops.index(acc) if acc is not None else None}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
       if key not in self.load_cache:
         if acc is not None:
-          self.load_cache[key] = self.uops.add(UOps.DEFINE_ACC, localtype, ((barrier,) if barrier else ()), (self.get_reduce_acc(acc), i, acc_count))
+          id = self.reduceops.index(acc)*2 + (0 if self.bufs[i].__class__ is LocalBuffer else 1)
+          self.load_cache[key] = self.uops.add(UOps.DEFINE_ACC, localtype, (), (self.get_reduce_acc(acc), id, acc_count))
```

Comment:
set "i" in a way that doesn't break with multiple accs

Reply:
I think id should be a combination of {reduceop_idx} + {early acc or late acc}.
eg:
https://github.com/Qazalin/tinygrad/commit/2a7b707d2594491e154195d8e9620e6f652a6698#diff-63d57daaca59e5495bbb108da1c13e782f4a408db8503e015e8094ac07c87ea6R109
you could also something like https://github.com/Qazalin/tinygrad/commit/c44261fb4da9ca1a7ad1483d6fe0afa7c4308b43
I think string is simpler to reason about.

Reply:
I like the string too, the change is in this diff: https://github.com/tinygrad/tinygrad/pull/4259/commits/57125675cebbfc0bc3aa8325e6fab4a3e22e1e56